assets/src/edit-story/app/media/local/test/useContextValueProvider.js

```diff
@@ -0,0 +1,263 @@
+      },
+      full: {
+        file: 'IMAGE.jpg',
+        width: 1080,
+        height: 2220,
+        mime_type: 'image/jpeg',
+        source_url: 'http://wp.local/wp-content/uploads/2020/09/IMAGE.jpg',
+      },
+    },
+    attribution: undefined,
+  },
+];
+
+describe('useContextValueProvider', () => {
+  let reducerState;
+  let reducerActions;
+  beforeEach(() => {
+    reducerState = {
+      processing: [],
+      processed: [],
+      media: [],
+      pageToken: '',
+      mediaType: '',
+      searchTerm: '',
+    };
+    reducerActions = {
+      fetchMediaStart: jest.fn(),
+      fetchMediaSuccess: jest.fn(),
+      fetchMediaError: jest.fn(),
+      resetFilters: jest.fn(),
+      setMedia: jest.fn(),
+      setMediaType: jest.fn(),
+      setSearchTerm: jest.fn(),
+      setNextPage: jest.fn(),
+      setProcessing: jest.fn(),
+      removeProcessing: jest.fn(),
+      updateMediaElement: jest.fn(),
+      deleteMediaElement: jest.fn(),
+    };
+
+    const useUploadMediaResult = {
+      uploadMedia: jest.fn(),
+      isUploading: jest.fn(),
+    };
+    useUploadMedia.mockImplementation(() => useUploadMediaResult);
+
+    const useUploadVideoFrameResult = {
+      uploadVideoFrame: jest.fn(),
+    };
+    useUploadVideoFrame.mockImplementation(() => useUploadVideoFrameResult);
+  });
+
+  it('resetWithFetch calls getMedia with cacheBust:true and then fetchMediaSuccess', async () => {
+    // Set up and make initial call to useContextValueProvider (which calls
+    // getMedia and fetchMediaSuccess once). Sets an empty media state.
+    const getMedia = jest.fn(() =>
+      Promise.resolve({
+        data: [],
+        headers: GET_MEDIA_RESPONSE_HEADER,
+      })
+    );
+    const apiState = {
+      actions: {
+        getMedia,
+      },
+    };
+    const configState = {
+      api: {},
+      allowedMimeTypes: {
+        video: [],
+      },
+    };
+    const { result } = renderHook(
+      () => useContextValueProvider(reducerState, reducerActions),
+      {
+        // eslint-disable-next-line react/display-name
+        wrapper: (params) => (
+          <ConfigProvider config={configState}>
+            <ApiContext.Provider value={apiState}>
+              {params.children}
+            </ApiContext.Provider>
+          </ConfigProvider>
+        ),
+      }
+    );
+
+    // This promise will only complete when the "done()" callback is called
+    // (see reducerActions.fetchMediaSuccess mock implementation in Promise).
+    await new Promise((done) => {
+      getMedia.mockImplementation(() => {
+        return Promise.resolve({
+          data: GET_MEDIA_RESPONSE_BODY,
+          headers: GET_MEDIA_RESPONSE_HEADER,
+        });
+      });
+      reducerActions.fetchMediaSuccess.mockImplementation(() => {
+        done();
+      });
+
+      // Act:
+      result.current.actions.resetWithFetch();
+    });
+
+    // Assert after fetchMediaSuccess callback is called:
```

Comment:
Swap the expect statements, to reflect the order in which fetchMediaSuccess then getMedia are called.

Reply:
getMedia get's called first and then the fetchMediaSuccess get's called.

See `local/useContextValueProvider:84`. The "callback" is the fetchMediaSuccess function which is called after getMedia's promise is resolved via "then".

Reply:
I came to this conclusion after reading "Assert after fetchMediaSuccess callback is called".

Reply:
Hmm yeah thats a confusing/redundant statement. Removed.